src/ray/common/task/scheduling_resources.cc

```diff
@@ -227,29 +229,48 @@ void ResourceSet::AddResources(const ResourceSet &other) {
   }
 }
 
-void ResourceSet::AddBundleResources(const PlacementGroupID &bundle_id,
+void ResourceSet::AddBundleResources(const PlacementGroupID &group_id,
                                      const int bundle_index, const ResourceSet &other) {
   for (const auto &resource_pair : other.GetResourceAmountMap()) {
+    // With bundle index (e.g., CPU_group_i_zzz).
     const std::string &resource_label =
-        FormatPlacementGroupResource(resource_pair.first, bundle_id, bundle_index);
+        FormatPlacementGroupResource(resource_pair.first, group_id, bundle_index);
     const FractionalResourceQuantity &resource_capacity = resource_pair.second;
     resource_capacity_[resource_label] += resource_capacity;
+
+    // Without bundle index (e.g., CPU_group_zzz).
+    const std::string &wildcard_label =
+        FormatPlacementGroupResource(resource_pair.first, group_id, -1);
+    resource_capacity_[wildcard_label] += resource_capacity;
   }
 }
 
-void ResourceSet::ReturnBundleResources(const PlacementGroupID &bundle_id,
+void ResourceSet::ReturnBundleResources(const PlacementGroupID &group_id,
                                         const int bundle_index) {
+  absl::flat_hash_map<std::string, FractionalResourceQuantity> to_restore;
   for (auto iter = resource_capacity_.begin(); iter != resource_capacity_.end();) {
     const std::string &bundle_resource_label = iter->first;
-    if (bundle_resource_label.find(bundle_id.Hex()) != std::string::npos) {
+    // We only consider the indexed resources, ignoring the wildcard resource.
+    // This is because when multiple bundles are created on one node, the quantity
+    // of the wildcard resources contains resources from multiple bundles.
+    if (IsBundleIndex(bundle_resource_label, group_id, bundle_index)) {
       const std::string &resource_label = GetOriginalResourceName(bundle_resource_label);
       const FractionalResourceQuantity &resource_capacity = iter->second;
-      resource_capacity_[resource_label] += resource_capacity;
+      to_restore[resource_label] = resource_capacity;
       iter = resource_capacity_.erase(iter);
     } else {
       iter++;
     }
   }
+  // For each matching resource to restore (e.g., key like CPU, GPU).
+  for (const auto &pair : to_restore) {
+    resource_capacity_[pair.first] += pair.second;
+    auto wildcard_resource = FormatPlacementGroupResource(pair.first, group_id, -1);
```

Comment:
Add the default parameter -1 to the function declaration of FormatPlacementGroupResource?

Reply:
Done.